File: 프로그래머스/2/42842. 카펫/카펫.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    // w >= h
    
    // w -> 모든 약수 ? (>=3)
    // h = total / w
    
    // (w-2) * (h-2) = yellow
    let total: Int = brown + yellow
    var candidates: [Int] = []
    var rst: [Int] = []
    
    // find width cand
    for i in 3...total {
        if total % i == 0 {
            candidates.append(i)
        }
    }
    
    for w in candidates {
        let h = total / w
        
        if (w-2) * (h-2) == yellow {
            rst = [w,h]
        }
    }
    
    return rst
    
    
    
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const totalArea = brown + yellow;
    
    for(let h =3; h <= Math.sqrt(totalArea); h++){
        if(totalArea % h === 0){
 			const w = totalArea / h;           
            
            if((h-2) * (w-2) === yellow){
                return [w,h];
            }
        }
    }
}